sdcm/utils/cdc/gce_builder.py

```diff
@@ -0,0 +1,250 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright (c) 2022 ScyllaDB
+import os
+import sys
+import logging
+from typing import Any
+from functools import cached_property
+
+import click
+import requests
+from google.oauth2 import service_account
+from google.cloud import compute_v1
+from google.api_core.extended_operation import ExtendedOperation
+import google.api_core.exceptions
+
+from sdcm.utils.gce_region import GceRegion
+from sdcm.utils.gce_utils import SUPPORTED_PROJECTS
+from sdcm.sct_runner import GceSctRunner
+from sdcm.utils.sct_cmd_helpers import CloudRegion, add_file_logger
+from sdcm.keystore import KeyStore
+
+LOGGER = logging.getLogger(__name__)
+
+JENKINS_CONFIG_FORMAT = """
+// just modify this config other code just logic
+config = [
+    name: "{name}",
+    region: "{region_name}",
+    projectId: "{project_id}",
+    credentialsId: "{project_id}",
+    instanceCapStr: "{max_instances}",
+    labels: "{jenkins_labels}",
+]
+
+import com.google.jenkins.plugins.computeengine.ComputeEngineCloud
+import com.google.jenkins.plugins.computeengine.InstanceConfiguration
+import com.google.jenkins.plugins.computeengine.SshConfiguration
+import hudson.plugins.sshslaves.SSHConnector
+import hudson.plugins.sshslaves.verifiers.NonVerifyingKeyVerificationStrategy
+import hudson.model.*
+import jenkins.model.Jenkins
+
+ComputeEngineCloud gcpCloud = new ComputeEngineCloud(config.name, config.projectId, config.credentialsId, config.instanceCapStr)
+gcpCloud.setNoDelayProvisioning(true)
+
+InstanceConfiguration instanceConfiguration = new InstanceConfiguration()
+instanceConfiguration.setNumExecutorsStr("1")
+instanceConfiguration.setOneShot(true)
+instanceConfiguration.setNamePrefix("builders")
+instanceConfiguration.setDescription("SCT Jenkins builders")
+instanceConfiguration.setRegion("https://www.googleapis.com/compute/v1/projects/{project_id}/regions/{region_name}")
+instanceConfiguration.setZone("https://www.googleapis.com/compute/v1/projects/{project_id}/zones/{region_name}-b")
+instanceConfiguration.setLabelString(config.labels)
+instanceConfiguration.setMode(Node.Mode.EXCLUSIVE)
+instanceConfiguration.setTemplate("https://www.googleapis.com/compute/v1/projects/{project_id}/global/instanceTemplates/{name}")
+instanceConfiguration.setRetentionTimeMinutesStr("6")
+instanceConfiguration.setLaunchTimeoutSecondsStr("300")
+instanceConfiguration.setRunAsUser("jenkins")
+instanceConfiguration.setSshConfiguration(SshConfiguration.builder().customPrivateKeyCredentialsId("6e17c350-0a47-4a59-b0b9-7e64f1c35a65").build())
+
+gcpCloud.addConfiguration(instanceConfiguration)
+
+// get Jenkins instance
+Jenkins jenkins = Jenkins.get()
+
+// clear old configuration
+jenkins.clouds.removeAll {{ it.name == "gce-${{config.name}}" }}
+
+res = jenkins.clouds.add(gcpCloud)
+
+// save current Jenkins state to disk
+jenkins.save()
+"""
+
+
+class GceBuilder:
```

Comment:
please add some doc related this class responsibility

Reply:
added the description